pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -336,6 +339,58 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 			swInfo)
 	}
 
+	// Reporting all zpools in Strorage Info
+	if vault.ReadPersistType() == types.PersistZFS {
+		zfsVersion, err := zfs.GetZfsVersion()
+		if err != nil {
+			log.Errorf("error: %v", err)
+		}
+
+		zpoolList, err := libzfs.PoolOpenAll()
+		if err != nil {
+			log.Errorf("get zpool list failed %v", err)
+		} else {
+			for _, zpool := range zpoolList {
+				rStorageInfo := new(info.StorageInfo)
+				curentRaid := info.StorageRaidType_STORAGE_RAID_TYPE_NORAID
+				zpoolName := zpool.Properties[libzfs.PoolPropName].Value
+				zpoolSizeInByte, err := strconv.ParseUint(zpool.Properties[libzfs.PoolPropSize].Value, 10, 64)
+				if err != nil {
+					log.Errorf("error with ParseUint for get zpool size in byte: %v", err)
+					zpoolSizeInByte = 0
+				}
+
+				dedup, _ := strconv.ParseFloat(zpool.Properties[libzfs.PoolPropDedupratio].Value, 64)
```

Comment:
I hope we can use `compressratio` of persist dataset. In my experiments:
```
9f021aea-8c01-4f84-9514-3dbd433efbda:~# zpool get dedupratio persist
NAME     PROPERTY    VALUE  SOURCE
persist  dedupratio  1.00x  -
```
```
9f021aea-8c01-4f84-9514-3dbd433efbda:~# zfs get compressratio persist
NAME     PROPERTY       VALUE  SOURCE
persist  compressratio  1.70x  -
```

Reply:
Corrected